crates/rpc/src/module.rs

```diff
@@ -0,0 +1,182 @@
+            // why info here? it's the same used in warp tracing filter for example.
+            let span = tracing::info_span!("rpc_method", name = method_name);
+            async move {
+                method((*context).clone()).await.map_err(|err| {
+                    let rpc_err: RpcError = err.into();
+                    jsonrpsee::core::Error::from(rpc_err)
+                })
+            }
+            .instrument(span)
+        };
+
+        self.0
+            .register_async_method(method_name, method_callback)
+            .with_context(|| format!("Registering {method_name}"))?;
+
+        Ok(self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::RpcContext;
+    use crate::error::RpcError;
+    use crate::test_client::TestClientBuilder;
+    use jsonrpsee::http_server::HttpServerBuilder;
+    use jsonrpsee::rpc_params;
+    use std::net::{Ipv4Addr, SocketAddr, SocketAddrV4};
+
+    #[tokio::test]
+    async fn without_input() {
+        let ctx = RpcContext::for_tests();
+
+        async fn say_hello(_: RpcContext) -> Result<String, RpcError> {
+            Ok("hello".to_string())
+        }
+
+        let methods = super::Module::new(ctx)
+            .register_method_with_no_input("say_hello", say_hello)
+            .unwrap()
+            .build();
+
+        let server = HttpServerBuilder::default()
+            .build(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, 0)))
+            .await
+            .unwrap();
+        let addr = server.local_addr().unwrap();
+
+        let _jh = server.start(methods).unwrap();
+
+        let client = TestClientBuilder::default()
+            .request_timeout(std::time::Duration::from_secs(2))
+            .address(addr)
+            .build()
+            .unwrap();
+
+        let message = client.request::<String>("say_hello", None).await.unwrap();
+        assert_eq!(message.as_str(), "hello");
+    }
+
+    #[tokio::test]
+    async fn with_input() {
+        use jsonrpsee::rpc_params;
+
+        let ctx = RpcContext::for_tests();
+
+        async fn echo(_: RpcContext, input: String) -> Result<String, RpcError> {
+            Ok(input)
+        }
+
+        let methods = super::Module::new(ctx)
+            .register_method("echo", echo)
+            .unwrap()
+            .build();
+
+        let server = HttpServerBuilder::default()
+            .build(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::LOCALHOST, 0)))
+            .await
+            .unwrap();
+        let addr = server.local_addr().unwrap();
+
+        let _jh = server.start(methods).unwrap();
+
+        let client = TestClientBuilder::default()
+            .request_timeout(std::time::Duration::from_secs(2))
+            .address(addr)
+            .build()
+            .unwrap();
+
+        let message = client
+            .request::<String>("echo", rpc_params!["testing testing 123"])
+            .await
+            .unwrap();
+        assert_eq!(message, "testing testing 123");
+    }
```

Comment:
@CHr15F0x as mentioned offline, I can't get this test to work correctly. I'd appreciate some kung-fu here.

Reply:
Thanks 🙏